### ISSUE TITLE
Shortcuts: a way forward?

### ISSUE THREAD

**Issue by ellatrix** [WordPress/gutenberg#15190]
Shortcuts have always been a bit problematic:

* they can be remapped by the user,
* they may conflict with the OS, browser, or assistive technologies,
* and some keys may not be available by the user, as keyboard layouts changes based on the OS and language.

Examples:

* `Ctrl+B` may change depending on the language.
* `Ctrl+Alt` combinations may conflict with  assistive technologies, see #11154.
* How does `Ctrl+A` work for keyboards without an `A` key?
* `Ctrl+D` is used as an alternative to `Delete`, see #15019.
* `Ctrl+Y` may or may not be implemented as redo, depending on the OS and browser, see #8921.

Solution: we need to look at _intention_ instead of trying to interpret key combinations.

Key interpretation must be removed if it includes at least one key that is not universal. For example: `A`, `B`, `C`...
In editable areas (`textarea` and `contenteditable`), the [`InputEvent.inputType`](https://w3c.github.io/input-events/#interface-InputEvent-Attributes) can be used. For example: `formatBold` means the user wants to format something bold.
The problem is that this cannot be used for block selection. When a modifier key is pressed (`Ctrl`, `Cmd` or `Alt/Option`), we could insert an invisible `textarea` with two characters and place the caret in the middle. Attach `input` and `selectionChange` event listeners. With these we can see if the user wants to select all content (check selection), undo or redo something (`historyUndo`, `historyRedo`), or delete something (`deleteContentForward`). If the modifier key is released again, restore focus to the block wrapper. 

There one thing that is hard to resolve however: documentation for these shortcuts. We are completely unaware which key combination would be the right one for e.g. bolding text. We could make the combination text translatable though, and note in the shortcut modal that it may not be accurate if your OS language is different or if you have mapped the shortcuts.

Thoughts? Cc @afercia.

**Comment by afercia** [comment]
Related: #15335, #3218 (October 2017).

I'm not fully sure I understand how `inputType` can be used to avoid keyboard shortcuts conflicts. Probably I'm missing something 🙂 

Re: the translations issue, I'd just note that historically most of the keyboard shortcuts on Windows don't change depending on the language. I guess that's to make them consistent across languages. Also, I'm guessing (operating systems aren't really my area of expertise) that the responsibility to change keyboard shortcuts depending on the language is up to the single applications. For example, see below the Windows Notepad, in English and Italian:

![notepad English](https://user-images.githubusercontent.com/1682452/57064117-b3a80180-6cc5-11e9-828c-6b9365aa3665.png)

![notepad Italian](https://user-images.githubusercontent.com/1682452/57064119-b3a80180-6cc5-11e9-8e54-cd4fcbd6e6f6.png)

Most fo the keyboard shortcuts don't change. Only "Find" does.
